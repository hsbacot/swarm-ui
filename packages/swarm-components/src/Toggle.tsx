import * as React from 'react';
import { Icon } from './Icon';

export type Props = {
	/**
	 * Indicated whether the toggle is selected
	 */
	checked: boolean,
	/**
	 * Indicated whether the toggle is disabled
	 */
	disabled?: boolean,
	/**
	 * Action to be performed when the toggle is interacted with
	 */
	onChange?: () => null,
};

const FILLS = {
	checked: 'var(--color-viridian)',
	unchecked: 'var(--color-gray-4)',
};

const Toggle = (props: Props): React.ReactElement => {
	const { checked = false, disabled = false, ...rest } = props;
	const checkedStatus = checked ? 'checked' : 'unchecked';
	const fillIcon = checked ? 'check' : 'cross';

	return (
		<button
			data-swarm-toggle={checkedStatus}
			role="checkbox"
			type="button"
			aria-checked={checked}
			aria-readonly={disabled}
			disabled={disabled}
			{...rest}
		>
			<span data-swarm-toggle-switch-disc>
				<Icon shape={fillIcon} />
			</span>
		</button>
	);
};

export { Toggle };
