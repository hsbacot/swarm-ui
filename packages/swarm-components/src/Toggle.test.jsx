import React from 'react';
import { shallow } from 'enzyme';
import { Toggle } from './Toggle';

describe('Toggle', () => {
	const testCases = [
		['Default', <Toggle key="1" />],
		['Checked', <Toggle key="2" checked />],
		['Disabled', <Toggle key="3" disabled />],
		['Disabled Checked', <Toggle key="4" disabled checked />],
	];

	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
	});
});
