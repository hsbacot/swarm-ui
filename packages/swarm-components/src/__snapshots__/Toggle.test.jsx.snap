// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshot: Checked 1`] = `
<button
  aria-checked={true}
  aria-readonly={false}
  data-swarm-toggle="checked"
  disabled={false}
  role="checkbox"
  type="button"
>
  <span
    data-swarm-toggle-switch-disc={true}
  >
    <Icon
      shape="check"
    />
  </span>
</button>
`;

exports[`Toggle Snapshot: Default 1`] = `
<button
  aria-checked={false}
  aria-readonly={false}
  data-swarm-toggle="unchecked"
  disabled={false}
  role="checkbox"
  type="button"
>
  <span
    data-swarm-toggle-switch-disc={true}
  >
    <Icon
      shape="cross"
    />
  </span>
</button>
`;

exports[`Toggle Snapshot: Disabled 1`] = `
<button
  aria-checked={false}
  aria-readonly={true}
  data-swarm-toggle="unchecked"
  disabled={true}
  role="checkbox"
  type="button"
>
  <span
    data-swarm-toggle-switch-disc={true}
  >
    <Icon
      shape="cross"
    />
  </span>
</button>
`;

exports[`Toggle Snapshot: Disabled Checked 1`] = `
<button
  aria-checked={true}
  aria-readonly={true}
  data-swarm-toggle="checked"
  disabled={true}
  role="checkbox"
  type="button"
>
  <span
    data-swarm-toggle-switch-disc={true}
  >
    <Icon
      shape="check"
    />
  </span>
</button>
`;
